### PR TITLE
feat: NotValidVar for checking invalid valid when getting var

### PR DESCRIPTION
**Pull request by lekaha**

### CODE REVIEW

**Comment by lekaha** [comment]
@pokk 
I think they are our goal:
- We need `default constructor` for using ModelMapper
- Giving `default value` for all constructor parameters if using Data class
- But we need `specialized default` value as invalid value 

Would like to discuss about our DAO class with Delegates Properties. I think to achieve this, our choices are
- Using `normal class` with Delegates Properties to `check the value` is valid or not.
- Using `data class` with Invalid Default Value but `cannot be checked`.
- Using 'data class` with Delegates Properties but it makes more code required. (https://github.com/lekaha/playone-android/pull/69/files#diff-efec1ecfaa9115d0a7353120bcece90eR14)

What do you think?

**Comment by lekaha** [comment]
@pokk If no problem please merge

File: ext/src/main/java/com/playone/mobile/common/NotValidVar.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import kotlin.reflect.KProperty
+
+class NotValidVar<T : Any>(private val invalidVal: T) : ReadWriteProperty<Any?, T> {
+    private var value: T = invalidVal
+
+    public override fun getValue(thisRef: Any?, property: KProperty<*>): T =
+        if (value != invalidVal)
+            value
+        else
+            throw IllegalStateException("Property ${property.name} should be initialized before get.")
```
Comment (pokk) [inline ext/src/main/java/com/playone/mobile/common/NotValidVar.kt]: The msg is wrong. It should be `"Property ${property.name} value is invalid, please set a valid value."` or something blah blah blah.

Comment (pokk) [reply ext/src/main/java/com/playone/mobile/common/NotValidVar.kt]: Also here shouldn't throw an exception for an invliad value I think.

**Review by pokk — CHANGES_REQUESTED** [review]

**Comment by lekaha** [comment]
@pokk Please check again

File: ext/src/main/java/com/playone/mobile/common/NotValidVar.kt (hunk 1 above)
Comment (lekaha) [reply ext/src/main/java/com/playone/mobile/common/NotValidVar.kt]: Ok, it should not be RuntimeException

File: ext/src/main/java/com/playone/mobile/common/NotValidVar.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        if (value != invalidVal)
+            value
+        else {
+            InvalidPropertyValueException("Property ${property.name} is setting invalid value")
+                .printStackTrace()
+            invalidVal
+        }
+
+    override fun setValue(thisRef: Any?, property: KProperty<*>, value: T) {
+        this.value = value
```
Comment (pokk) [inline ext/src/main/java/com/playone/mobile/common/NotValidVar.kt]: `this.value = value` should be next line.

**Review by pokk — APPROVED** [review]

**Pull request merged**